### ISSUE TITLE
ClientRequest: write callback get called twice (`FormData` are broken)

### ISSUE THREAD

**Issue by mikicho** [mswjs/interceptors#684]
The regression is introduced in `0.36.10`. When the interceptor is applied, we call the write callback twice.
```js
const https = require('https');
const { ClientRequestInterceptor } = require('@mswjs/interceptors/ClientRequest')

const int = new ClientRequestInterceptor()
int.apply()

const req = https.request('https://webhook.site/ae950c30-b7bf-46e7-bb40-2c32ca132689', {
  method: 'POST',
},
  res => {
    console.log(res.statusCode);
    res.resume();
  })


req.write('mamma mia', null, () => {
  console.log(1); // prints 1 twice
  req.end()
})
```

----

From what I see, for some reason, Node calls to the callback as well in https://github.com/nodejs/node/blob/3fb2ea8689250a3d39d9c66afd7301c7caa644c2/lib/internal/streams/writable.js#L764

I'm not sure why tho.

**Comment by kettanaito** [comment]
Thanks for debugging this, @mikicho. Can you confirm that exhausting the write buffer indeed fixes the FormData issue?

There was some context behind us leaving the buffer intact. The issues were #659 and https://github.com/mswjs/msw/issues/2309, both concerning a passthrough scenario. In the fix, I removed the `writeBuffer` exhaustion (unshift) altogether for all types of requests.

Perhaps the right way to tackle this is to:

1. Not exhaust the `writeBuffer` (keep as is).
2. Call the write callbacks in `MockHttpSocket.flushWriteBuffer` _but_ don't call the callbacks that were already called in `passthrough()`. 

We can do the latter by either settings the called callbacks to undefined (`writeBufferEntry[2] = undefined`) in the loop, or by setting a custom symbol on the called callbacks and ignoring them in the `passthrough` method. 

Which one sounds the best to you?

**Comment by kettanaito** [comment]
The issue you are describing is also only reproducible for requests that end inside the `.write()` call. The write callback indeed gets called once if you do

```ts
req.write('')
req.end()
```

**Comment by kettanaito** [comment]
@mikicho, I've opened a fix for this in https://github.com/mswjs/interceptors/pull/685. Your reproduction test above is now passing. Could you please review my take on the issue when you have a moment?

**Comment by kettanaito** [comment]
## Released: v0.37.2 🎉

This has been released in v0.37.2!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.37.2)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.37.2)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
